### ISSUE TITLE
`config list` Not Showing Global `target-dev-hub` value

### ISSUE THREAD

**Issue by RCMoonpie** [forcedotcom/cli#2435]
### Summary
After configuring a local value via `config set --target-dev-hub`, and also a global value for `--target-dev-hub`, executing `sf config list` only shows the Local.

### Steps To Reproduce
- Execute `sf config set target-dev-hub=ALIAS`
   ```
   Name            Value  Success
   --------------  -----  -------
   target-dev-hub  ALIAS  true
   ```
- Execute `sf config list`
   ```
   Name            Location  Value
   target-dev-hub  Local     ALIAS
   ```
- Execute `sf config set --global target-dev-hub=ALIAS`
   ```
   Name            Value  Success
   --------------  -----  -------
   target-dev-hub  ALIAS  true
   ```

### Expected result
Execute `sf config list`
   ```
   Name            Location  Value
   target-dev-hub  Global    ALIAS
   target-dev-hub  Local     ALIAS
   ```

### Actual result
Execute `sf config list`
   ```
   Name            Location  Value
   target-dev-hub  Local     ALIAS
   ```

### System Information
Using Windows VS Code 1.81.0 Powershell 5.1 terminal.

```json
{
  "cliVersion": "@salesforce/cli/2.5.8",
  "architecture": "win32-x64",
  "nodeVersion": "node-v18.17.0",
  "osVersion": "Windows_NT 10.0.19045",
  "shell": "cmd.exe",
  "rootPath": "C:\\Programs Portable\\node-v18.17.0-win-x64\\node_modules\\@salesforce\\cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.6 (core)",
    "@oclif/plugin-commands 2.2.22 (core)",
    "@oclif/plugin-help 5.2.17 (core)",
    "@oclif/plugin-not-found 2.3.37 (core)",
    "@oclif/plugin-plugins 3.2.6 (core)",
    "@oclif/plugin-search 0.0.22 (core)",
    "@oclif/plugin-update 3.1.32 (core)",
    "@oclif/plugin-version 1.3.8 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.48 (core)",
    "@oclif/plugin-which 2.2.30 (core)",
    "@salesforce/cli 2.5.8 (core)",
    "apex 2.3.10 (core)",
    "auth 2.8.12 (core)",
    "data 2.5.6 (core)",
    "deploy-retrieve 1.17.5 (core)",
    "info 2.6.39 (core)",
    "limits 2.3.30 (core)",
    "login 1.2.26 (core)",
    "org 2.10.0 (core)",
    "schema 2.3.23 (core)",
    "settings 1.4.25 (core)",
    "sobject 0.2.4 (core)",
    "source 2.10.31 (core)",
    "telemetry 2.3.0 (core)",
    "templates 55.5.10 (core)",
    "trust 2.6.1 (core)",
    "user 2.3.28 (core)"
 ]
}
```

### Additional information
- The `--help` for `config list` states (emphasis added)
"Global configuration variables apply to any Salesforce DX project and are **always** displayed. If you run this command from a project directory, local configuration variables are **also** displayed."

  *It is understood that local configurations are used over global ones. However, it is immensely helpful to know if you have a global configuration set at all -- whether it is the same or a different value.*

- Occurred originally on `sf cli 2.2.7`. Updated to latest stable release of `2.5.8` and the same thing occurred.

- Searched open & closed issues using `global config` and `"config list"`, but saw nothing that seemed to be the same.  (Closest was [2426](https://github.com/forcedotcom/cli/issues/2426#issue-1864686858), but it does not appear to be the same.)

**Comment by shetzel** [comment]
This is working as designed in terms of the command behavior, which displays the config values that are applied based on the current context.  We will update the docs to make it more clear.

**Comment by shetzel** [comment]
@RCMoonpie - if we added a `--all` flag to this command and displayed all config settings, would that be helpful?  Would you use it?

**Comment by jshackell-sfdc** [comment]
This issue is fixed in CLI version [2.14.6 (Oct 25, 2023)](https://github.com/forcedotcom/cli/blob/main/releasenotes/README.md#2146-oct-25-2023-stable).
